lci_strategic_plugin/src/lci_strategic_plugin.cpp

```diff
@@ -753,191 +760,95 @@ void LCIStrategicPlugin::planWhenAPPROACHING(const cav_srvs::PlanManeuversReques
-  }
+    if (in_tbd) // Given ET is in TBD, but vehicle is too close to intersection
+    {
+      ROS_DEBUG_STREAM("ET is still in TBD despite the vehicle being in desired distance to start stopping. Trying to handle this edge case gracefully...");
+    }
 
-  ROS_DEBUG_STREAM("Final nearest_green_entry_time: " << std::to_string(nearest_green_entry_time.toSec()));
+    double stopping_time = current_state.speed / 1.5 * max_comfort_decel_norm_; //one half the acceleration (twice the acceleration to stop) to account for emergency case
 
-  auto et_state = traffic_light->predictState(lanelet::time::timeFromSec(nearest_green_entry_time.toSec()));
-  ROS_DEBUG_STREAM("Signal at ET: " << et_state.get().second);
+    ros::Time stopping_arrival_time =
+          current_state.stamp + ros::Duration(stopping_time);
 
-  double remaining_time = nearest_green_entry_time.toSec() - current_state.stamp.toSec();
-  double remaining_time_earliest_entry = earliest_entry_time.toSec() - current_state.stamp.toSec();
-  scheduled_entry_time_ = remaining_time; // performance metric
-  earliest_entry_time_ = remaining_time_earliest_entry; // performance metric
-  
-  // CASE SELECTION START
-  auto boundary_distances = get_delta_x(current_state_speed, intersection_speed_.get(), speed_limit, config_.algo_minimum_speed, max_comfort_accel_, max_comfort_decel_);
-  print_boundary_distances(boundary_distances); //debug
+    ROS_DEBUG_STREAM("stopping_arrival_time: " << std::to_string(stopping_arrival_time.toSec()));
 
-  auto boundary_traj_params = get_boundary_traj_params(req.header.stamp.toSec(), current_state_speed, intersection_speed_.get(), speed_limit, config_.algo_minimum_speed, max_comfort_accel_, max_comfort_decel_, current_state.downtrack, traffic_light_down_track, distance_remaining_to_traffic_light, boundary_distances);
+    auto stopping_arrival_state_optional = traffic_light->predictState(lanelet::time::timeFromSec(stopping_arrival_time.toSec()));
 
-  TrajectoryParams ts_params = get_ts_case(req.header.stamp.toSec(), nearest_green_entry_time.toSec(), current_state_speed, intersection_speed_.get(), speed_limit, config_.algo_minimum_speed, max_comfort_accel_, max_comfort_decel_, current_state.downtrack, traffic_light_down_track, distance_remaining_to_traffic_light, boundary_distances, boundary_traj_params);
-  print_params(ts_params);
+    if (!validLightState(stopping_arrival_state_optional, stopping_arrival_time))
+    {
+      ROS_ERROR_STREAM("Unable to resolve give signal for stopping_arrival_state_optional: " << std::to_string(stopping_arrival_time.toSec()));
+      return;
+    }
 
-  ROS_DEBUG_STREAM("SPEED PROFILE CASE:" << ts_params.case_num);
+    if (stopping_arrival_state_optional.get().second == lanelet::CarmaTrafficSignalState::STOP_AND_REMAIN || 
+      stopping_arrival_state_optional.get().second == lanelet::CarmaTrafficSignalState::PROTECTED_CLEARANCE)
```

Comment:
After more discussions, I feel like maybe we can get rid of yellow change in this _if_ condition. Since we are checking the stopping arrival time, maybe we don't want the vehicle to stop if the stopping arrival time is within yellow!!!

Reply:
gotcha removed.